server/src/main/scala/org/ada/server/dataaccess/dataset/DataSetAccessor.scala

```diff
@@ -59,6 +60,7 @@ protected class DataSetAccessorImpl(
     dataSetSettingRepo: DataSetSettingRepo
   ) extends DataSetAccessor {
 
+  private val dataSetUpdateSemaphore = new Semaphore(1) // couldn't use ReentrantLock here because a single thread must lock and unlock the lock, which doesn't hold in 'future'
```

Comment:
When passing constants like this it would make sense to assign it to a variable or used named parameters to improve readability

Reply:
with any other value this code wouldn't work so it's not a "tweakable" / adjustable constant per se